crates/fluvio-cli/src/client/topic/create.rs

```diff
@@ -105,28 +135,72 @@ pub struct CreateTopicOpt {
 impl CreateTopicOpt {
     pub async fn process(self, fluvio: &Fluvio) -> Result<()> {
         let dry_run = self.dry_run;
-        let (name, topic_spec) = self.construct()?;
+        let admin = fluvio.admin().await;
+        let (name, topic_spec) = self.construct(&admin).await?;
         validate(&name, &topic_spec)?;
 
         debug!("creating topic: {} spec: {:#?}", name, topic_spec);
-        let admin = fluvio.admin().await;
         admin.create(name.clone(), dry_run, topic_spec).await?;
         println!("topic \"{name}\" created");
 
         Ok(())
     }
 
-    fn construct(self) -> Result<(String, TopicSpec)> {
+    async fn construct(self, admin: &FluvioAdmin) -> Result<(String, TopicSpec)> {
         if let Some(config_path) = self.config {
             let config = TopicConfig::from_file(config_path)?;
             return Ok((config.meta.name.clone(), config.into()));
         }
 
+        let topic_name = self.topic.unwrap_or_default();
```

Comment:
topic name can't be empty. it should be passed from previously

Reply:
Even for Assigned topics? I didn't change this behavior.

Reply:
Ok.  Let's fix this logic in another PR